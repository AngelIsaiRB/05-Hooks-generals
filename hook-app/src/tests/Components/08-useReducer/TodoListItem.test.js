import { shallow } from "enzyme"
import { TodoListItem } from "../../../Components/08-useReducer/TodoListItem"
import { demoTodos } from "../../fixtures/demoTodos"

const demoData= demoTodos;

describe('pruebas en el componente TodoListItem', () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(<TodoListItem todo={demoData[0]} index={demoTodos[0].id} handleDelete={handleDelete} handleToggle={handleToggle}/>);
    test('should de mostrarse correctamente ', () => {
       
        expect(wrapper).toMatchSnapshot();
    });


    test('should de llmar la funcion borrar', () => {
        wrapper.find("button").simulate("click");
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
    });

    test('should de llmar la funcion borrar', () => {
        wrapper.find("p").simulate("click");
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
    });

    test('should de llmar la funcion toggle', () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(`2. ${demoTodos[0].desc}`)
    });

    test('should de teber la clase complete si done esta en true ', () => {
        const todo= demoTodos[0];
        todo.done=true;
        const wrapper = shallow(<TodoListItem todo={demoData[0]}/>);
        expect(wrapper.find("p").hasClass("complete")).toBe(true);
    });
    
    
    
})
