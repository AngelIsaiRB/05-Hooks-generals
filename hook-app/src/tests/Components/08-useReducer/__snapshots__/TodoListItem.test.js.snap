// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente TodoListItem should de mostrarse correctamente  1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1"
  >
    <p
      className="false"
      onClick={[Function]}
    >
       
      2
      . 
      aprender react
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Borrar
    </button>
  </li>
</Fragment>
`;
